public/app/features/alerting/unified/components/AlertLabels.tsx

```diff
@@ -1,17 +1,41 @@
+import { css } from '@emotion/css';
+import { chain } from 'lodash';
 import React from 'react';
 
-import { TagList } from '@grafana/ui';
+import { GrafanaTheme2 } from '@grafana/data';
+import { getTagColorsFromName, useStyles2 } from '@grafana/ui';
+
+import { Label } from './Label';
 
 interface Props {
   labels: Record<string, string>;
-  className?: string;
+  size?: 'md' | 'sm';
 }
 
-export const AlertLabels = ({ labels, className }: Props) => {
-  const pairs = Object.entries(labels).filter(([key]) => !(key.startsWith('__') && key.endsWith('__')));
+export const AlertLabels = ({ labels, size }: Props) => {
+  const styles = useStyles2((theme) => getStyles(theme, size));
+  const pairs = chain(labels).toPairs().reject(isPrivateKey).value();
+
   return (
-    <div className={className}>
-      <TagList tags={Object.values(pairs).map(([label, value]) => `${label}=${value}`)} className={className} />
+    <div className={styles.wrapper} role="list" aria-label="Labels">
+      {pairs.map(([label, value]) => (
+        <Label key={label + value} size={size} label={label} value={value} color={getLabelColor(label)} />
+      ))}
     </div>
   );
 };
+
+function getLabelColor(input: string): string {
+  return getTagColorsFromName(input).color;
+}
+
+const isPrivateKey = ([key, _]: [string, string]) => key.startsWith('__') && key.endsWith('__');
+
+const getStyles = (theme: GrafanaTheme2, size?: string) => ({
```

Comment:
maybe in here we should define size as :` size?: 'md' | 'sm'`

Reply:
Good point, I should abstract this in to a separate union type